### DISCUSSION TITLE
Using batch loaders with the new virtual resolvers

### DISCUSSION THREAD

**Discussion by ericuldall** [feathersjs/feathers#3338] [Q&A] [answered]
It seems that in  v5 our fastJoin has been upgraded to virtual resolvers. However, I believe these are susceptible to the same performance issues we saw with fastJoin.

## Example
I query 1000 documents with `{ userId: 1234 }` and I have a virtual resolver that does `userService.get(userId)` so I now have 1000 db lookups for a single user record that should really just be one lookup per event lifecycle.

Previously we used batch loaders to cache common lookups in a single request and speed up our api / reduce overhead on our db/network. https://hooks-common.feathersjs.com/guides.html#using-batch-loaders

Is there a new solution to address this problem in v5? Do we still use batch loaders? I'm curious what the latest greatest solution are for this.

Another note, I'm not interested (necessarily) in running a full scale caching solution per service, but I like the idea of temporarily caching results during the life of a single request. I'm open to convincing that full scale service caching is the way though.

Any thoughts on this?

**Comment by daffl** [comment, marked as answer]
The [feathers-dataloader](https://github.com/feathersjs-ecosystem/dataloader) works well with resolvers and combine simultaneous requests into one.

**Reply by ericuldall** [reply]
I'm trying it out. Not sure if it's production ready, see my issues
https://github.com/feathersjs-ecosystem/dataloader/issues/7
https://github.com/feathersjs-ecosystem/dataloader/issues/6
